Headers/AppKit/NSCollectionView.h

```diff
@@ -44,47 +58,384 @@ enum
+shouldSelectItemsAtIndexPaths: (NSSet *)indexPaths;
+
+/**
+ * Returns the set of NSIndexPath objects that should be deselected given the proposed indexPaths.
+ */
+- (NSSet *) collectionView: (NSCollectionView *)collectionView shouldDeselectItemsAtIndexPaths: (NSSet *)indexPaths;
+
+/**
+ * Called to inform the delegate of those items that were selected.
+ */
+- (void) collectionView: (NSCollectionView *)collectionView didSelectItemsAtIndexPaths: (NSSet *)indexPaths;
+
+/**
+ * Called to inform the delegate of those items that were deselected.
+ */
+- (void) collectionView: (NSCollectionView *)collectionView didDeselectItemsAtIndexPaths: (NSSet *)indexPaths;
+
+/* Display Notification */
+
+/**
+ * Called to inform the delegate that the item representing the object at indexPath will be displayed.
+ */
+- (void) collectionView: (NSCollectionView *)collectionView
+        willDisplayItem: (NSCollectionViewItem *)item
+        forRepresentedObjectAtIndexPath: (NSIndexPath *)indexPath;
+
+/**
+ * Called to inform the delegate that the supplementary view for the elementKind will be displayed 
+ * at indexPath.
+ */
+- (void) collectionView: (NSCollectionView *)collectionView
+         willDisplaySupplementaryView: (NSView *)view
+         forElementKind: (NSCollectionViewSupplementaryElementKind)elementKind
+            atIndexPath: (NSIndexPath *)indexPath;
+
+/**
+ * Called to inform the delegate that the collectionView will end display of item for the object at indexPath.
+ */
+- (void) collectionView: (NSCollectionView *)collectionView
+   didEndDisplayingItem: (NSCollectionViewItem *)item
+   forRepresentedObjectAtIndexPath: (NSIndexPath *)indexPath;
+
+/**
+ * Called to inform the delegate that the collectionView will end display of supplementaryView for the 
+ * object at indexPath.
+ */
+- (void) collectionView: (NSCollectionView *)collectionView
+         didEndDisplayingSupplementaryView: (NSView *)view
+       forElementOfKind: (NSCollectionViewSupplementaryElementKind)elementKind
+            atIndexPath: (NSIndexPath *)indexPath;
+
+/* Layout Transition Support */
+
+/**
+ * Called to inform the delegate that the collectionView is transitioning from the old to the new
+ * layout indicated.
+ */
+- (NSCollectionViewTransitionLayout *) collectionView: (NSCollectionView *)collectionView
+                         transitionLayoutForOldLayout: (NSCollectionViewLayout *)fromLayout
+                                            newLayout: (NSCollectionViewLayout *)toLayout;
+#endif
+@end
 
 APPKIT_EXPORT_CLASS
 @interface NSCollectionView : NSView //<NSDraggingDestination, NSDraggingSource>
 {
+  // Content
   NSArray *_content;
   IBOutlet NSCollectionViewItem *itemPrototype;
   NSMutableArray *_items;
-  
+
+  // Selection management...
   BOOL _allowsMultipleSelection;
   BOOL _isSelectable;
+  BOOL _allowsEmptySelection;
   NSIndexSet *_selectionIndexes;
+  NSSet *_selectionIndexPaths;
```

Comment:
No, one is for NSIndexPaths.   I believe that they are fundamentally different.   If there is a way to store both in the same place, that would be better, but I did it this way because I wasn't sure if I could store the indexPaths in the selectionIndexes ivar.

Reply:
I'm going to resolve this one.